src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -29,6 +31,12 @@ public static void Build(Context context, Document file)
 
             var outputPath = context.DocumentProvider.GetOutputPath(file.FilePath);
 
+            var jsonOutputPath = LegacyUtility.ChangeExtension(context.DocumentProvider.GetOutputPath(file.FilePath, true), ".json");
+            if (string.Compare(jsonOutputPath, metadata.BreadcrumbPath, System.StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Yes, good catch. I think I mixed up validations with a different rule. I added the check in `LoadTocNode` since `GetHrefType` is static. Please verify.